fabric-api-base/src/main/java/net/fabricmc/fabric/api/util/TriMap.java

```diff
@@ -0,0 +1,52 @@
+package net.fabricmc.fabric.api.util;
+
+import java.util.Objects;
+import java.util.Set;
+
+import com.google.common.collect.Sets;
+
+public class TriMap<K, L, R> {
```

Comment:
How much does this differ from Table and why is this in the api?

Reply:
It's basically a 3d map

Reply:
It was supposed to be used for flammable blocks but the vanilla way was for flammable blocks far easier

Reply:
that's what the table is, 3 dimensional thing